// redux/reducers/registerReducer.js
const REGISTER_USER = 'REGISTER_USER'
const IN_SYSTEM = 'IN_SYSTEM'
const REGISTER_DONE = 'REGISTER_DONE'
const LOGIN_TOKEN = 'LOGIN_TOKEN'
const UPDATE_EMAIL = 'UPDATE_EMAIL'
const UPDATE_USERNAME = 'UPDATE_USERNAME'
const ADD_ID = 'ADD_ID'

const initialState = {
    registerUser: {
        name: '',
        email: '',
        password: '',
        role: ''
    },
    
    registerDone: {
        success: false,
        token: ''
    },

    inSystem: false,
    loginToken: '',

    updateUserName: '',
    updateUserEmail: '',
    id: ''
}

function registerReducer (state = initialState, action) {
    switch (action.type) {
        case REGISTER_USER: return {
            ...state,
            registerUser: action.payload
        }
        case IN_SYSTEM: return  {
            ...state,
            inSystem: action.payload
        }
        case REGISTER_DONE: return {
            ...state,
            registerDone: action.payload
        }
        case LOGIN_TOKEN: return {
            ...state,
            loginToken: action.payload
        }
        case UPDATE_EMAIL: return {
            ...state,
            updateUserEmail: action.payload
        }
        case UPDATE_USERNAME: return {
            ...state,
            updateUserName: action.payload
        }
        case ADD_ID: return {
            ...state,
            id: action.payload
        }
        default: return state
    }
}

export default registerReducer
